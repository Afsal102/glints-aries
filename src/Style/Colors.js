export const PrimaryColor = {
  glintsred: '#EC272B',
  glintsblue: '#0BAEEC',
  glintsyellow: '#FFF240',
};

export const SecondaryColor = {
  white: '#FFFFFF',
  whitesmoke: '#F5F5F5',
  black: '#000000',
  lightblack: '#777777',
  grey: '#646464',
  lightgrey: '#C6C6C6',
  buttercup: '#F5A623',
  purplepizza: '#FF00E5',
  blue: '#0869C2',
  bluegem: '#9013FE',
};

