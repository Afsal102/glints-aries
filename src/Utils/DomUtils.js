// DOM utilities

/**
 * Returns a value when ESC button is triggered.
 *
 * @param  {Boolean} value
 * @return {Boolean}
 */
export function escEvent(value) {
  const listener = (e) => {
    if (e.keyCode === 27) {
      return value();
    }
    return 0;
  };

  return listener;
}
