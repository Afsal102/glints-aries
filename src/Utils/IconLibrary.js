import React from 'react';

export const ICONS = {
    ADD: <path d="M368.5 240H272v-96.5c0-8.8-7.2-16-16-16s-16 7.2-16 16V240h-96.5c-8.8 0-16 7.2-16 16 0 4.4 1.8 8.4 4.7 11.3 2.9 2.9 6.9 4.7 11.3 4.7H240v96.5c0 4.4 1.8 8.4 4.7 11.3 2.9 2.9 6.9 4.7 11.3 4.7 8.8 0 16-7.2 16-16V272h96.5c8.8 0 16-7.2 16-16s-7.2-16-16-16z"/>,
    ADD_OUTLINE: <path d="M346.5 240H272v-74.5c0-8.8-7.2-16-16-16s-16 7.2-16 16V240h-74.5c-8.8 0-16 6-16 16s7.5 16 16 16H240v74.5c0 9.5 7 16 16 16s16-7.2 16-16V272h74.5c8.8 0 16-7.2 16-16s-7.2-16-16-16z M256 76c48.1 0 93.3 18.7 127.3 52.7S436 207.9 436 256s-18.7 93.3-52.7 127.3S304.1 436 256 436c-48.1 0-93.3-18.7-127.3-52.7S76 304.1 76 256s18.7-93.3 52.7-127.3S207.9 76 256 76m0-28C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"/>,
    ARROW_BACK: <path d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"/>,
    ARROW_DOWN: <path d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"/>,
    ARROW_NEXT: <path d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"/>,
    ARROW_UP: <path d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"/>,
    CAMERA: <g><circle cx="256" cy="275" r="57.5"/><path d="M417.5 160H363c-4.6 0-8.9-2-12-5.4-28.4-31.8-39.1-42.6-50.7-42.6h-85.5c-11.7 0-23.2 10.8-51.7 42.7-3 3.4-7.4 5.3-11.9 5.3h-4.1v-8c0-4.4-3.6-8-8-8h-26c-4.4 0-8 3.6-8 8v8h-7.5C79.9 160 64 173.2 64 190.7v176c0 17.5 15.9 33.3 33.5 33.3h320c17.6 0 30.5-15.8 30.5-33.3v-176c0-17.5-12.9-30.7-30.5-30.7zM260 360.4c-50.3 2.3-91.7-39.1-89.4-89.4 2-43.9 37.5-79.4 81.4-81.4 50.3-2.3 91.7 39.1 89.4 89.4-2 43.9-37.5 79.4-81.4 81.4zM352 218c-7.2 0-13-5.8-13-13s5.8-13 13-13 13 5.8 13 13-5.8 13-13 13z"/></g>,
    CHECKMARK: <path d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"/>,
    CHECKMARK_OUTLINE: <path d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z M256 76c48.1 0 93.3 18.7 127.3 52.7S436 207.9 436 256s-18.7 93.3-52.7 127.3S304.1 436 256 436c-48.1 0-93.3-18.7-127.3-52.7S76 304.1 76 256s18.7-93.3 52.7-127.3S207.9 76 256 76m0-28C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"/>,
    CLOSE: <path d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"/>,
    CLOSE_OUTLINE: <path d="M331.3 308.7L278.6 256l52.7-52.7c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-52.7-52.7c-6.2-6.2-15.6-7.1-22.6 0-7.1 7.1-6 16.6 0 22.6l52.7 52.7-52.7 52.7c-6.7 6.7-6.4 16.3 0 22.6 6.4 6.4 16.4 6.2 22.6 0l52.7-52.7 52.7 52.7c6.2 6.2 16.4 6.2 22.6 0 6.3-6.2 6.3-16.4 0-22.6z M256 76c48.1 0 93.3 18.7 127.3 52.7S436 207.9 436 256s-18.7 93.3-52.7 127.3S304.1 436 256 436c-48.1 0-93.3-18.7-127.3-52.7S76 304.1 76 256s18.7-93.3 52.7-127.3S207.9 76 256 76m0-28C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"/>,
    CONTACT: <path d="M256 48C148.5 48 60.1 129.5 49.2 234.1c-.8 7.2-1.2 14.5-1.2 21.9 0 7.4.4 14.7 1.2 21.9C60.1 382.5 148.5 464 256 464c114.9 0 208-93.1 208-208S370.9 48 256 48zm135.8 326.1c-22.7-8.6-59.5-21.2-82.4-28-2.4-.7-2.7-.9-2.7-10.7 0-8.1 3.3-16.3 6.6-23.3 3.6-7.5 7.7-20.2 9.2-31.6 4.2-4.9 10-14.5 13.6-32.9 3.2-16.2 1.7-22.1-.4-27.6-.2-.6-.5-1.2-.6-1.7-.8-3.8.3-23.5 3.1-38.8 1.9-10.5-.5-32.8-14.9-51.3-9.1-11.7-26.6-26-58.5-28h-17.5c-31.4 2-48.8 16.3-58 28-14.5 18.5-16.9 40.8-15 51.3 2.8 15.3 3.9 35 3.1 38.8-.2.7-.4 1.2-.6 1.8-2.1 5.5-3.7 11.4-.4 27.6 3.7 18.4 9.4 28 13.6 32.9 1.5 11.4 5.7 24 9.2 31.6 2.6 5.5 3.8 13 3.8 23.6 0 9.9-.4 10-2.6 10.7-23.7 7-58.9 19.4-80 27.8C91.6 341.4 76 299.9 76 256c0-48.1 18.7-93.3 52.7-127.3S207.9 76 256 76c48.1 0 93.3 18.7 127.3 52.7S436 207.9 436 256c0 43.9-15.6 85.4-44.2 118.1z"/>,
    FACEBOOK: <path d="M426.8 64H85.2C73.5 64 64 73.5 64 85.2v341.6c0 11.7 9.5 21.2 21.2 21.2H256V296h-45.9v-56H256v-41.4c0-49.6 34.4-76.6 78.7-76.6 21.2 0 44 1.6 49.3 2.3v51.8h-35.3c-24.1 0-28.7 11.4-28.7 28.2V240h57.4l-7.5 56H320v152h106.8c11.7 0 21.2-9.5 21.2-21.2V85.2c0-11.7-9.5-21.2-21.2-21.2z"/>,
    INSTAGRAM: <path d="M336 96c21.2 0 41.3 8.4 56.5 23.5S416 154.8 416 176v160c0 21.2-8.4 41.3-23.5 56.5S357.2 416 336 416H176c-21.2 0-41.3-8.4-56.5-23.5S96 357.2 96 336V176c0-21.2 8.4-41.3 23.5-56.5S154.8 96 176 96h160m0-32H176c-61.6 0-112 50.4-112 112v160c0 61.6 50.4 112 112 112h160c61.6 0 112-50.4 112-112V176c0-61.6-50.4-112-112-112z M360 176c-13.3 0-24-10.7-24-24s10.7-24 24-24c13.2 0 24 10.7 24 24s-10.8 24-24 24zM256 192c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64m0-32c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96z"/>,
    LINKEDIN: <path d="M417.2 64H96.8C79.3 64 64 76.6 64 93.9V415c0 17.4 15.3 32.9 32.8 32.9h320.3c17.6 0 30.8-15.6 30.8-32.9V93.9C448 76.6 434.7 64 417.2 64zM183 384h-55V213h55v171zm-25.6-197h-.4c-17.6 0-29-13.1-29-29.5 0-16.7 11.7-29.5 29.7-29.5s29 12.7 29.4 29.5c0 16.4-11.4 29.5-29.7 29.5zM384 384h-55v-93.5c0-22.4-8-37.7-27.9-37.7-15.2 0-24.2 10.3-28.2 20.3-1.5 3.6-1.9 8.5-1.9 13.5V384h-55V213h55v23.8c8-11.4 20.5-27.8 49.6-27.8 36.1 0 63.4 23.8 63.4 75.1V384z"/>,
    NOTIFICATION: <path d="M289.7 403c-6.1 0-11.4 4.2-12.7 10.2-1 4.5-2.7 8.2-5 10.9-1.3 1.5-5.1 5.9-16.1 5.9-11 0-14.8-4.5-16.1-5.9-2.3-2.7-4-6.4-5-10.9-1.3-6-6.6-10.2-12.7-10.2-8.4 0-14.5 7.8-12.7 15.9 5 22.3 21 37.1 46.5 37.1s41.5-14.7 46.5-37.1c1.8-8.1-4.4-15.9-12.7-15.9zM412 352.2c-15.4-20.3-45.7-32.2-45.7-123.1 0-93.3-41.2-130.8-79.6-139.8-3.6-.9-6.2-2.1-6.2-5.9v-2.9c0-13.3-10.8-24.6-24-24.6h-.6c-13.2 0-24 11.3-24 24.6v2.9c0 3.7-2.6 5-6.2 5.9-38.5 9.1-79.6 46.5-79.6 139.8 0 90.9-30.3 102.7-45.7 123.1-9.9 13.1-.5 31.8 15.9 31.8h280.1c16.1 0 25.4-18.8 15.6-31.8zm-39 5.8H139.8c-3.8 0-5.8-4.4-3.3-7.3 7-8 14.7-18.5 21-33.4 9.6-22.6 14.3-51.5 14.3-88.2 0-37.3 7-66.5 20.9-86.8 12.4-18.2 27.9-25.1 38.7-27.6 8.4-2 14.4-5.8 18.6-10.5 3.2-3.6 8.7-3.8 11.9-.2 5.1 5.7 12 9.1 18.8 10.7 10.8 2.5 26.3 9.4 38.7 27.6 13.9 20.3 20.9 49.5 20.9 86.8 0 36.7 4.7 65.6 14.3 88.2 6.5 15.2 14.4 25.9 21.5 33.9 2.2 2.7.4 6.8-3.1 6.8z"/>,
    PERSON: <path d="M447.8 438.3c-7.2-31.8-48.3-47.3-62.5-52.3-15.6-5.5-37.8-6.8-52.1-10-8.2-1.8-20.1-6.3-24.1-11.1s-1.6-49.3-1.6-49.3 7.4-11.5 11.4-21.7c4-10.1 8.4-37.9 8.4-37.9s8.2 0 11.1-14.4c3.1-15.7 8-21.8 7.4-33.5-.6-11.5-6.9-11.2-6.9-11.2s6.1-16.7 6.8-51.3c.9-41.1-31.3-81.6-89.6-81.6-59.1 0-90.6 40.5-89.7 81.6.8 34.6 6.7 51.3 6.7 51.3s-6.3-.3-6.9 11.2c-.6 11.7 4.3 17.8 7.4 33.5 2.8 14.4 11.1 14.4 11.1 14.4s4.4 27.8 8.4 37.9c4 10.2 11.4 21.7 11.4 21.7s2.4 44.5-1.6 49.3c-4 4.8-15.9 9.3-24.1 11.1-14.3 3.2-36.5 4.5-52.1 10-14.2 5-55.3 20.5-62.5 52.3-1.1 5 2.7 9.7 7.9 9.7H440c5.1 0 8.9-4.7 7.8-9.7z"/>,
    TWITTER: <path d="M492 109.5c-17.4 7.7-36 12.9-55.6 15.3 20-12 35.4-31 42.6-53.6-18.7 11.1-39.4 19.2-61.5 23.5C399.8 75.8 374.6 64 346.8 64c-53.5 0-96.8 43.4-96.8 96.9 0 7.6.8 15 2.5 22.1-80.5-4-151.9-42.6-199.6-101.3-8.3 14.3-13.1 31-13.1 48.7 0 33.6 17.2 63.3 43.2 80.7-16-.4-31-4.8-44-12.1v1.2c0 47 33.4 86.1 77.7 95-8.1 2.2-16.7 3.4-25.5 3.4-6.2 0-12.3-.6-18.2-1.8 12.3 38.5 48.1 66.5 90.5 67.3-33.1 26-74.9 41.5-120.3 41.5-7.8 0-15.5-.5-23.1-1.4C62.8 432 113.7 448 168.3 448 346.6 448 444 300.3 444 172.2c0-4.2-.1-8.4-.3-12.5C462.6 146 479 129 492 109.5z"/>
};