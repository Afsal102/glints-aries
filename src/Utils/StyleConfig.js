export const Variant = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  GHOST: 'ghost',
  LINK: 'link',
};

export const Color = {
  RED: 'red',
  YELLOW: 'yellow',
  BLUE: 'blue',
};