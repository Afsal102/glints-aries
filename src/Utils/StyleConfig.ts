export enum ETabAlignment {
  HORIZONTAL = 'horizontal',
  VERTICAL = 'vertical',
}

export enum EHorizontalTabVariant {
  UNDERLINED = 'underlined',
  COLORED = 'colored',
}

export enum ETabThemeVariant {
  BLUE = 'blue',
  BLACK = 'black',
}

export enum ETooltipPosition {
  TOP = 'top',
  BOTTOM = 'bottom',
  LEFT = 'left',
  RIGHT = 'right',
}

export const ButtonVariant = {
  SOLID_WHITE: 'solid-white',
  SOLID_BLUE: 'solid-blue',
  YELLOW: 'yellow',
  GHOST: 'ghost',
  WHITE_GREY: 'white-grey',
  LINK: 'link',
};

export const PsychedelicTheme = {
  BLUE_DOMINANT: 'blue-dominant',
  BLUE_DEFAULT: 'blue-default',
  RED_DOMINANT: 'red-dominant',
  RED_DEFAULT: 'red-default',
  YELLOW_DOMINANT: 'yellow-dominant',
  YELLOW_DEFAULT: 'yellow-default',
};

export const ButtonTheme = {
  BLUE: 'blue',
  WHITE: 'white',
};

export const ScreenSize = {
  mobileS: 320,
  mobileM: 480,
  mobileL: 640,
  tablet: 768,
  desktopS: 1024,
  desktopM: 1260,
  desktopL: 1440,
};

export const Device = {
  mobileS: `(max-width: ${ScreenSize.mobileM - 1}px)`,
  mobileM: `(min-width: ${ScreenSize.mobileM}px)`,
  mobileL: `(min-width: ${ScreenSize.mobileL}px)`,
  tablet: `(min-width: ${ScreenSize.tablet}px)`,
  desktopS: `(min-width: ${ScreenSize.desktopS}px)`,
  desktopM: `(min-width: ${ScreenSize.desktopM}px)`,
  desktopL: `(min-width: ${ScreenSize.desktopL}px)`,
};
