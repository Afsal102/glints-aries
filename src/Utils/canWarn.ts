export const canWarn =
  typeof console !== 'undefined' && process.env.NODE_ENV === 'development';
