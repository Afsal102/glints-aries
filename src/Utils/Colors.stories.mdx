import { ColorItem, ColorPalette, Meta } from '@storybook/addon-docs';
import { PrimaryColor, Greyscale, SecondaryColor } from './Colors';
import { getReadableColor } from './getReadableColor';
import { ColorSampler, ColorSample } from './ColorsStoriesHelpers';
import blue from './@next/colors/blue';
import green from './@next/colors/green';
import neutral from './@next/colors/neutral';
import orange from './@next/colors/orange';
import red from './@next/colors/red';
import yellow from './@next/colors/yellow';

<Meta title="Utilities/Colors" />

# Color

```
import { PrimaryColor, Greyscale } from 'glints-aries';

const MyComponent = styled.div`
  background-color: ${PrimaryColor.glintsyellow};
  color: ${PrimaryColor.grey};
`;
```

## PrimaryColor

<ColorSampler>
  {Object.keys(PrimaryColor).map(colorName => (
    <ColorSample
      key={colorName}
      bgColor={PrimaryColor[colorName]}
      fontColor={getReadableColor(PrimaryColor[colorName])}
      name={colorName}
    />
  ))}
</ColorSampler>

## Greyscale

<ColorSampler>
  {Object.keys(Greyscale).map(colorName => (
    <ColorSample
      key={colorName}
      bgColor={Greyscale[colorName]}
      fontColor={getReadableColor(Greyscale[colorName])}
      name={colorName}
    />
  ))}
</ColorSampler>

## SecondaryColor

<ColorSampler>
  {Object.keys(SecondaryColor).map(colorName => (
    <ColorSample
      key={colorName}
      bgColor={SecondaryColor[colorName]}
      fontColor={getReadableColor(SecondaryColor[colorName])}
      name={colorName}
    />
  ))}
</ColorSampler>

# @next

## Brand Colors

<ColorSampler>
  <ColorSample key={yellow.Brand} bgColor={yellow.Brand} name="yellow.Brand" />
  <ColorSample key={red.Brand} bgColor={red.Brand} name="red.Brand" />
  <ColorSample key={blue.Brand} bgColor={blue.Brand} name="blue.Brand" />
  <ColorSample key={orange.Brand} bgColor={orange.Brand} name="orange.Brand" />
</ColorSampler>

## Product Colors

<ColorPalette>
  <ColorItem
    title="blue"
    subtitle="blue.codeName"
    colors={Object.fromEntries(Object.entries(blue).slice(1, 7))}
  />
  <ColorItem title="green" subtitle="green.codeName" colors={green} />
  <ColorItem
    title="neutral whites"
    subtitle="neutral.codeName"
    colors={Object.fromEntries(Object.entries(neutral).slice(0, 5))}
  />
  <ColorItem
    title="neutral blacks"
    subtitle="neutral.codeName"
    colors={Object.fromEntries(Object.entries(neutral).slice(5, 10))}
  />
  <ColorItem
    title="orange"
    subtitle="orange.codeName"
    colors={Object.fromEntries(Object.entries(orange).slice(1, 6))}
  />
  <ColorItem
    title="red"
    subtitle="red.codeName"
    colors={Object.fromEntries(Object.entries(red).slice(1, 6))}
  />
  <ColorItem
    title="yellow"
    subtitle="yellow.codeName"
    colors={Object.fromEntries(Object.entries(yellow).slice(1, 6))}
  />
</ColorPalette>
