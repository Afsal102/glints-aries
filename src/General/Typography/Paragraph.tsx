import * as React from 'react';
import classNames from 'classnames';

import {
  Paragraph as StyledParagraph,
  ParagraphProps,
} from './ParagraphStyles';
import { SecondaryColor } from '../../Utils/Colors';

const Paragraph: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    variant = 'regular',
    bold,
    color = SecondaryColor.black,
    ellipsis,
    ...restProps
  } = props;

  return (
    <StyledParagraph
      className={classNames('aries-typography-paragraph', className)}
      variant={variant}
      bold={bold}
      color={color}
      ellipsis={ellipsis}
      {...restProps}
    >
      {children}
    </StyledParagraph>
  );
};

interface Props extends ParagraphProps {
  className?: string;
  children: React.ReactNode;
}

export default Paragraph;
