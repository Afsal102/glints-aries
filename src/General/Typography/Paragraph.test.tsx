import * as React from 'react';
import Paragraph from './Paragraph';
import { PrimaryColor } from '../../Utils/Colors';

import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import {
  PARAGRAPH_FONT_SIZES,
  PARAGRAPH_VARIANTS,
  paragraphType,
} from './ParagraphStyles';

const paragraphText =
  'As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent. They are continuously looking for dynamic, speed and competition, always being the first in everything - from work to social gatherings.';

describe('displays the correct text and font-size for each variant:', () => {
  const variants = Object.values(PARAGRAPH_VARIANTS) as paragraphType[];
  variants.forEach(variant => {
    const fontSize = PARAGRAPH_FONT_SIZES[variant];
    it(`${variant}: ${fontSize}px`, () => {
      const { getByText } = render(
        <Paragraph variant={variant}>{paragraphText}</Paragraph>
      );
      const paragraphTag = getByText(paragraphText);
      expect(paragraphTag).toBeTruthy();
      expect(paragraphTag).toHaveStyle(`font-size: ${fontSize}px`);
    });
  });
});

it('displays the text in bold', () => {
  const { getByText } = render(<Paragraph bold>{paragraphText}</Paragraph>);
  const paragraphTag = getByText(paragraphText);
  expect(paragraphTag).toHaveStyle('font-weight: bold');
});

it('displays the correct color', () => {
  const { getByText } = render(
    <Paragraph color={PrimaryColor.glintsred}>{paragraphText}</Paragraph>
  );
  const paragraphTag = getByText(paragraphText);
  expect(paragraphTag).toHaveStyle(`color: ${PrimaryColor.glintsred}`);
});

it('display the ellipsis', () => {
  const { getByText } = render(<Paragraph ellipsis>{paragraphText}</Paragraph>);
  const paragraphTag = getByText(paragraphText);
  expect(paragraphTag).toHaveStyle(`
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
  `);
});
