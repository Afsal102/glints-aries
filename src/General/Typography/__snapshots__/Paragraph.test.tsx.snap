// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the correct line height, when lineHeight value is set to true tag with lineHeight as true caption 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-size: 14px;
  line-height: 23.8px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
}

<p
    class="c0 aries-typography-paragraph"
    color="#000000"
  >
    As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent. They are continuously looking for dynamic, speed and competition, always being the first in everything - from work to social gatherings.
  </p>
</DocumentFragment>
`;

exports[`displays the correct line height, when lineHeight value is set to true tag with lineHeight as true regular 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-size: 16px;
  line-height: 27.2px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
}

<p
    class="c0 aries-typography-paragraph"
    color="#000000"
  >
    As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent. They are continuously looking for dynamic, speed and competition, always being the first in everything - from work to social gatherings.
  </p>
</DocumentFragment>
`;

exports[`displays the correct line height, when lineHeight value is set to true tag with lineHeight as true smallest 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-size: 12px;
  line-height: 20.4px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
}

<p
    class="c0 aries-typography-paragraph"
    color="#000000"
  >
    As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent. They are continuously looking for dynamic, speed and competition, always being the first in everything - from work to social gatherings.
  </p>
</DocumentFragment>
`;

exports[`displays the correct line height, when lineHeight value is set to true tag with lineHeight as true subtitle 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-size: 18px;
  line-height: 30.6px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
}

<p
    class="c0 aries-typography-paragraph"
    color="#000000"
  >
    As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent. They are continuously looking for dynamic, speed and competition, always being the first in everything - from work to social gatherings.
  </p>
</DocumentFragment>
`;
